Back-To-School/Assets/BackToSchool/Scripts/Player/PlayerController.cs

```diff
@@ -0,0 +1,166 @@
+            WeaponController.WeaponReloaded += Reload;
+            _playerInput.Fired              += Fire;
+            _playerInput.Stopped            += Stop;
+            _playerInput.Moved              += Move;
+            _playerInput.Rotated            += Rotate;
+            _playerInput.WeaponChanged      += NextWeapon;
+        }
+
+        private void Awake()
+        {
+            _rigidBody       = GetComponent<Rigidbody>();
+            _animator        = GetComponent<Animator>();
+            _renderers       = GetComponentsInChildren<SkinnedMeshRenderer>();
+            WeaponController = GetComponent<WeaponController>();
+            Inventory        = GetComponent<Inventory>();
+            WeaponController.SetInventory(Inventory);
+        }
+
+        private void OnDestroy()
+        {
+            _playerInput.Reloaded           -= WeaponController.Reload;
+            WeaponController.WeaponReloaded -= Reload;
+            _playerInput.Fired              -= Fire;
+            _playerInput.Stopped            -= Stop;
+            _playerInput.Moved              -= Move;
+            _playerInput.Rotated            -= Rotate;
+            _playerInput.WeaponChanged      -= NextWeapon;
+        }
+
+        #region Shooting
+
+        public void Reload()
+        {
+            if (!_isDead)
+                _animator.SetTrigger(Constants.AnimationStates.Reload.ToString());
+        }
+
+        public void ReloadFinished() => WeaponController.ReloadComplete();
+
+        public void Fire()
+        {
+            if (!_isDead)
+                WeaponController.Shoot(_playerStats.Damage.GetValue());
+        }
+
+        public void NextWeapon(bool isNext)
+        {
+            if (!_isDead)
+                WeaponController.NextWeapon(isNext);
+        }
+
+        #endregion
+
+        #region Interaction
+
+        public void InitializeHealth()
+        {
+            _currentHealth = _playerStats.MaxHealth.GetValue();
+            HealthChanged?.Invoke(_currentHealth);
+        }
+
+        public float GetHealthValue() => _currentHealth;
+
+        public void SetHealthValue(float health)
+        {
+            _currentHealth = health;
+            HealthChanged?.Invoke(_currentHealth);
+        }
+
+        public void TakeDamage(float damage)
+        {
+            damage         -= _playerStats.Armor.GetValue();
+            damage         =  Mathf.Clamp(damage, 1, int.MaxValue);
+            _currentHealth -= damage;
+            HealthChanged?.Invoke(_currentHealth);
+
+            if (_currentHealth <= 0 && !_isDead)
+            {
+                _animator.SetTrigger(Constants.AnimationStates.Die.ToString());
+                _isDead = true;
+                Died?.Invoke();
+            }
+            else if (_currentHealth > 0)
+                StartCoroutine(nameof(ShowDamageEffect));
+        }
+
+        private IEnumerator ShowDamageEffect()
```

Comment:
That ordering and grouping looks good, but ordering of public/private methods is broken :-(